src/HealthGPS/static_linear_model.cpp

```diff
@@ -85,6 +98,33 @@ void StaticLinearModel::linear_approximation(Person &person) {
     }
 }
 
+/// Initialises the weight of a person.
+///
+/// It uses the baseline adjustment to get its initial value, based on its sex and age.
+/// @param person The person fo initialise the weight for.
+/// @generator Random number generator for the simulation.
+void StaticLinearModel::initialise_weight(Person &person, Random generator) {
+
+    auto weight_bl = get_risk_factor_expected().at(person.gender, "Weight"_id).at(person.age);
+    auto weight_quantile = get_weight_quantile(person.gender, generator);
+    person.risk_factors["Weight"_id] = weight_bl * weight_quantile;
+}
+
+/// Returns the weight quantile for the given gender.
+///
+/// @gender The gender of the person.
+/// @generator Random number generator for the simulation.
+double StaticLinearModel::get_weight_quantile(core::Gender gender, Random generator) {
+
+    // TODO: Remove when fully implemented
+    if (weight_quantiles_.empty()) {
+        return 1.0;
+    }
+
+    int index = static_cast<int>(generator.next_double() * weight_quantiles_.at(gender).size());
```

Comment:
Furthermore, `Random` has a `next_int(int max_val)` method, so this can be:
```suggestion
    auto index = static_cast<size_t>(generator.next_int(weight_quantiles_.at(gender).size()));
```